### ISSUE TITLE
Bun exits "hastefully" when running via bun start, instead of waiting for SIGINT listeners

### ISSUE THREAD

**Issue by wundersolutions-juanjo** [oven-sh/bun#15878]
### What version of Bun is running?

1.1.40+b5b51004e

### What platform is your computer?

Microsoft Windows NT 10.0.19045.0 x64

### What steps can reproduce the bug?

Running `bun start`, with the following package.json and index.ts

package.json
```json
{
  "scripts": {
     "start": "bun ./index.ts"
   }
}
```

index.ts
```ts
process.on("SIGINT", () => {
  console.log("SIGTERM received, shutting down");
  setTimeout(() => process.exit(), 2000);
});

setInterval(() => {
  console.log("Running");
}, 500);
```

### What is the expected behavior?

Process gracefully exits via process.exit

### What do you see instead?

Process seems to be killed synchronously and instantly

**Comment by wundersolutions-juanjo** [comment]
The issue doesnt seem to be completely limited to bun, with weird behaviours when using npm or tsx


### Npm with bun runtime
```
PS C:\Users\Juanjo\Documents\git\test\issue> npm start

> start
> bun ./testFile.ts

SIGTERM received, shutting down
Running
Running
Running
Running
¿Desea terminar el trabajo por lotes (S/N)? n
PS C:\Users\Juanjo\Documents\git\test\issue>
```


### Npm with tsx runtime
```
PS C:\Users\Juanjo\Documents\git\test\issue> npm start

> start
> tsx ./testFile.ts

Running
SIGTERM received, shutting down
^CRunning
^C¿Desea terminar el trabajo por lotes (S/N)? Running                       <---- Caret stays here
Running
Running
```

### Bun with tsx
```
PS C:\Users\Juanjo\Documents\git\test\issue> bun start
$ tsx ./testFile.ts
Running
Running
SIGTERM received, shutting down
error: script "start" exited with code 58
PS C:\Users\Juanjo\Documents\git\test\issue> Running                   <---- Caret stays here
Running
Running
Running

```

**Comment by dy0gu** [comment]
Possibility related, I'm on Windows 11 and when running a Vite project with `bun` I get the following errors after every `Ctrl + C` on the Vite server.

`error: script "dev" exited with code 58` if I use `bun run dev` with this script:

```json
"scripts": {
	"dev": "vite",
},
```

&&

`error: "vite.exe" exited with code 58` if I use `bun run dev` with this script:

```json
"scripts": {
	"dev": "bunx --bun vite",
},
```

I didn't look thoroughly into this but I assume it's related since Vite probably tries to run some cleanup logic before the server is shutdown and so the error code is the same as mentioned in this issue.

It's probably also worth nothing that the second script method I mentioned, using `bunx --bun vite`, is the recommend in [this official guide](https://bun.sh/guides/ecosystem/vite) and I see no mention of expected errors there so I would assume this is not expected behavior with Vite.

**Comment by RiskyMH** [comment]
Yeah I think it was `bun run` closing instead of passing in the SIGINT message to the process. I believe `bun ./index.ts` or `bunx vite` should still work though.

**Comment by Saeid-Za** [comment]
I can confirm that this is a windows only problem.
But no matter how I'm starting the process:
- bun run index.ts 
- bun index.ts
- bun index.ts --watch
- bun run index.ts --watch
- build & bun dist/index.js
- bun & bun run dist/index.js


The behaviour is the same, **sometimes** the signal is not received in the app.
For me to reproduce the issue, jt takes 3-4 runs + CTRL c.
